nemo/collections/asr/helpers.py

```diff
@@ -163,3 +200,86 @@ def post_process_predictions(predictions, labels):
 
 def post_process_transcripts(transcript_list, transcript_len_list, labels):
     return __gather_transcripts(transcript_list, transcript_len_list, labels=labels)
+
+
+def process_classification_evaluation_batch(tensors: dict, global_vars: dict, top_k: list = 1):
+    """
+    Creates a dictionary holding the results from a batch of samples
+    """
+    if 'EvalLoss' not in global_vars.keys():
+        global_vars['EvalLoss'] = []
+    if 'batchsize' not in global_vars.keys():
+        global_vars['batchsize'] = []
+
+    if isinstance(top_k, int):
+        top_k = [top_k]
+
+    top_k = sorted(top_k)
+
+    for k in top_k:
+        if f'CorrectCount@{k}' not in global_vars.keys():
+            global_vars[f'CorrectCount@{k}'] = []
+
+    logits = None
+    labels = None
+
+    for kv, v in tensors.items():
+        if kv.startswith('loss'):
+            global_vars['EvalLoss'] += __gather_losses(v)
+        elif kv.startswith('logits'):
+            logits = torch.cat(v, 0)  # if len(v) > 1 else v
+        elif kv.startswith('label'):
+            labels = torch.cat(v, 0)  # if len(v) > 1 else v
+
+    batch_size = labels.size(0)
+    global_vars['batchsize'] += [batch_size]
+
+    with torch.no_grad():
+        topk_acc = classification_accuracy(logits, labels, top_k=top_k)
+
+    for k, acc in zip(top_k, topk_acc):
+        # Accuracy is provided as a percentage, we require the count of correct samples
+        # Therefore multiply by batch size to get count of correctly predicted samples
+        global_vars[f'CorrectCount@{k}'] += [acc * batch_size]
+
+
+def process_classification_evaluation_epoch(global_vars: dict, eval_metric=None, tag=None):
+    """
+    Calculates the aggregated loss and WER across the entire evaluation dataset
+    """
+    if eval_metric is None:
+        eval_metric = [1]
+
+    if type(eval_metric) not in (list, tuple):
+        eval_metric = [eval_metric]
+
+    top_k = eval_metric
+
+    eloss = torch.mean(torch.stack(global_vars['EvalLoss'])).item()
+    batch_sizes = global_vars['batchsize']
+    total_num_samples = torch.tensor(batch_sizes).sum().float()
+
+    topk_accs = []
+    for k in top_k:
+        correct_counts = torch.tensor(global_vars[f'CorrectCount@{k}'])
+        topk_acc = correct_counts.sum() / total_num_samples
+        topk_accs.append(topk_acc)
+
+    if tag is None:
```

Comment:
you can replace 
`tag='' if tag is None ` and use 

        logs = {f"Evaluation_Loss {tag}": eloss}
        logging.info(f"==========>>>>>>Evaluation Loss {tag}: {eloss}")
        for k, acc in zip(top_k, topk_accs):
            logging.info(f"==========>>>>>>Evaluation Accuracy Top@{k} {tag}: {acc * 100.:3.4f}")
            logs[f'Evaluation_Accuracy_Top@{k} {tag}'] = acc * 100.0

        return logs

without if else

Reply:
Fixed in latest commit